### ISSUE TITLE
How to use JsonBinaryType with Postgres and H2? 

### ISSUE THREAD

**Issue by Kurru** [vladmihalcea/hibernate-types#316]
I use Postgres in production and have a jsonb column. 

When talking to Postgres, 

```
@TypeDefs({
    @TypeDef(name = "json", typeClass = JsonStringType.class),
    @TypeDef(name = "jsonb", typeClass = JsonBinaryType.class)
})

...

@Type(type = "jsonb")
@Column(nullable = false, columnDefinition = "jsonb")
private String data;
```

Works as expected.

However, this causes issues in H2. Here, for my test to pass I need to configure: 
```
@Type(type = "json")
@Column(nullable = false, columnDefinition = "jsonb")
private String data;
```

Is there an approach to enable using the json type successfully in both Postgres and H2?

**Comment by vladmihalcea** [comment]
> Is there an approach to enable using the json type successfully in both Postgres and H2?

At the moment, there's no support for that, but it could be added. Is your company interested in sponsoring this feature?

**Comment by Kurru** [comment]
We are in the _extremely_ early stages of starting our company so sponsorships is something we've never considered before. I've sent you an email to learn more. 

So far I've got a partial solution following [this](https://stackoverflow.com/a/60821174/243203), but I've yet to find the final solution. Currently reading `"{\n  \"id\": \"some-id\"\n}"` from DB instead of `{"id": "some-id"}`.

**Comment by vladmihalcea** [comment]
Closed as duplicate of #174

The #174 will address the requirements of this issue as well.
